homeassistant/components/sensor/hive.py

```diff
@@ -0,0 +1,486 @@
+            return False
+        elif self.device_type == "HotWater_Mode":
+            return False
+        elif self.device_type == "HotWater_Boost":
+            return False
+        elif self.device_type == "Hive_Device_BatteryLevel":
+            return True
+        elif self.device_type == "Hive_Device_Sensor":
+            return True
+        elif self.device_type == "Hive_Device_Light_Mode":
+            return True
+        elif self.device_type == "Hive_Device_Plug_Mode":
+            return True
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        if self.device_type == "Heating_CurrentTemperature":
+            nodeid = self.node_id
+            curtempret = self.session.heating.current_temperature(nodeid)
+
+            if curtempret != -1000:
+                if nodeid in self.session.data.minmax:
+                    if (self.session.data.minmax[nodeid]['TodayDate'] !=
+                            datetime.date(datetime.now())):
+                        self.session.data.minmax[nodeid]['TodayMin'] = 1000
+                        self.session.data.minmax[nodeid]['TodayMax'] = -1000
+                        self.session.data.minmax[nodeid]['TodayDate'] = \
+                            datetime.date(datetime.now())
+
+                    if (curtempret < self.session.data.minmax[nodeid]
+                            ['TodayMin']):
+                        self.session.data.minmax[nodeid]['TodayMin'] = \
+                            curtempret
+
+                    if (curtempret > self.session.data.minmax[nodeid]
+                            ['TodayMax']):
+                        self.session.data.minmax[nodeid]['TodayMax'] = \
+                            curtempret
+
+                    if (curtempret < self.session.data.minmax[nodeid]
+                            ['RestartMin']):
+                        self.session.data.minmax[nodeid]['RestartMin'] = \
+                            curtempret
+
+                    if (curtempret >
+                            self.session.data.minmax[nodeid]
+                            ['RestartMax']):
+                        self.session.data.minmax[nodeid]['RestartMax'] = \
+                            curtempret
+                else:
+                    current_node_max_min_data = {}
+                    current_node_max_min_data['TodayMin'] = curtempret
+                    current_node_max_min_data['TodayMax'] = curtempret
+                    current_node_max_min_data['TodayDate'] = \
+                        datetime.date(datetime.now())
+                    current_node_max_min_data['RestartMin'] = curtempret
+                    current_node_max_min_data['RestartMax'] = curtempret
+                    self.session.data.minmax[nodeid] = \
+                        current_node_max_min_data
+            else:
+                curtempret = 0
+            return curtempret
+        elif self.device_type == "Heating_TargetTemperature":

```

Comment:
One solution would be to use a dict to map the device type to the `Session` attribute, ie `'heating'` or `'hotwater'` and method name. Then use `getattr` here after looking up the dictionary, to call the correct attribute method. But I'd rather try to make the interface of the pyhiveapi library more simple.

Reply:
i don't know how to do that. please can you help?